### PR TITLE
Move browser config to its own class

### PR DESCRIPTION
**Pull request by aellispierce**
This moves browser configuration to its own class and simplifies the `Driver` class to simply ask the browser for its `name`, `options` and `type`, instead of determining all of that within the driver class.

cc @eileencodes for 👀

### CODE REVIEW

File: actionpack/lib/action_dispatch/system_testing/browser.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module ActionDispatch
+  module SystemTesting
+    class Browser # :nodoc:
+      attr_reader :name
```
Comment (eileencodes) [inline actionpack/lib/action_dispatch/system_testing/browser.rb]: Can you add a new line under `attr_reader`?

File: actionpack/lib/action_dispatch/system_testing/browser.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      end
+
+      def type
+        if name == :headless_chrome
+          :chrome
+        elsif name == :headless_firefox
+          :firefox
+        else
+          name
+        end
```
Comment (eileencodes) [inline actionpack/lib/action_dispatch/system_testing/browser.rb]: This is picky, but I think this one and maybe the one below can be a case statement instead.

**Review by eileencodes — CHANGES_REQUESTED** [review]
Looking good, I added a few minor changes I'd like to see.

File: actionpack/lib/action_dispatch/system_testing/browser.rb (hunk 1 above)
Comment (aellispierce) [reply actionpack/lib/action_dispatch/system_testing/browser.rb]: Done in f52e17f

**Pull request merged**